code/game/objects/items/weapons/implants/implant_mindslave.dm

```diff
@@ -0,0 +1,71 @@
+	name = "mindslave implant"
+	desc = "Now YOU too can have your very own mindslave! Pop this implant into anybody and they'll obey any command you give for around 15 to 20 minutes."
+	origin_tech = "materials=2;biotech=4;programming=4"
+	activated = 0
+	var/timerid
+	var/datum/objective/protect/protect_objective
+
+/obj/item/weapon/implant/mindslave/get_data()
+	var/dat = {"<b>Implant Specifications:</b><BR>
+				<b>Name:</b> Syndicate Mindslave implant MK1<BR>
+				<b>Life:</b> Varies between 15 and 20 minutes.<BR>
+				<b>Important Notes:</b> Personnel injected with this device become loyal to the user and will obey any command given for a limited time.<BR>
+				<HR>
+				<b>Implant Details:</b><BR>
+				<b>Function:</b> Allows user to command anyone implanted to do whatever they want.<BR>
+				<b>Special Features:</b> Person with implant MUST obey any order you give. <BR>
+				<b>Integrity:</b> Implant will last around 15 and 20 minutes."}
+	return dat
+
+/obj/item/weapon/implant/mindslave/implant(mob/target,mob/user)
+	if(target == user)
+		target <<"<span class='notice'>You can't implant yourself!</span>"
+		return 0
+	if(isloyal(target))
+		target <<"<span class='danger'>Your loyalty implant rejects [user]'s mindslave!</span>"
+		user <<"<span class='danger'>[target] somehow rejects the mindslave implant!</span>"
+		return 0
+	if(..())
+		target << "<span class='notice'>You feel a surge of loyalty towards [user].</span>"
+		target << "<span class='userdanger'> You MUST obey any command given to you by your master(that doesn't violate any rules). You are an antag while mindslaved.</span>"
+		target << "<span class='danger'>You CANNOT harm your master. Check your memory ( with the notes verb) if you forget who your master is.</span>"
+		var/time = 9000 + rand(60,3000)
+		timerid = addtimer(src,"remove_mindslave",time)
+		if(!target.mind.special_role)
+			target.mind.special_role = "Mindslave"
+		protect_objective = new /datum/objective/protect
+		protect_objective.owner = target.mind
+		protect_objective.target = user.mind
+		protect_objective.explanation_text = "Protect [user], your mindslave master. Obey any command he gives."
+		target.mind.objectives += protect_objective
+		log_admin("[user]([user.ckey]) made a mindslave out of [target]([target.ckey]).")
+		return 1
+	return 0
+
+/obj/item/weapon/implant/mindslave/removed(mob/source)
+	deltimer(timerid)
+	remove_mindslave()
+	..()
+
+/obj/item/weapon/implant/mindslave/Destroy()
+	deltimer(timerid)
+	remove_mindslave()
+	..()
+
+/obj/item/weapon/implant/mindslave/proc/remove_mindslave()
+	if(imp_in)
+		if(imp_in.mind.special_role == "Mindslave")
+			imp_in.mind.special_role = ""
+		imp_in << "<span class='userdanger'>You feel your free will come back to you! You no longer have to obey your master!</span>"
+		imp_in << "<span class='userdanger'>If you were not an antagonist BEFORE being mindslave, then you no longer are one.</span>"
+		protect_objective = null
```

Comment:
No this is not how you do it, rather you should qdel it since it's a datum. Also test ingame and be sure that objectives list updates properly without nulls in it